fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/MetaDataProtoEditor.java

```diff
@@ -48,17 +49,16 @@
      *
      * <p>
      * Adding the record type involves three steps: the message type is added to the file descriptor's list of message types,
-     * a field of the given type is added to the union, and its primary key is set. This method takes
-     * care of the first two. Callers must also set the primary key
-     * ({@link com.apple.foundationdb.record.metadata.RecordTypeBuilder#setPrimaryKey} for the type.
+     * a field of the given type is added to the union, and its primary key is set.
      * Note that adding {@code UNION} record types is not allowed. To add {@code NESTED} record types, use {@link #addNestedRecordType}.
      * </p>
      *
      * @param metaDataBuilder the meta-data builder
      * @param newRecordType the new record type
+     * @param primaryKey the primary key of the new record type
      */
     @Nonnull
-    public static void addRecordType(@Nonnull RecordMetaDataProto.MetaData.Builder metaDataBuilder, @Nonnull DescriptorProtos.DescriptorProto newRecordType) {
+    public static void addRecordType(@Nonnull RecordMetaDataProto.MetaData.Builder metaDataBuilder, @Nonnull DescriptorProtos.DescriptorProto newRecordType, @Nonnull KeyExpression primaryKey) {
```

Comment:
I suppose in theory there should be a way to set the record type's `record_type_key` through this API (or a similar API?). Maybe it's fine to leave that as an issue for now and move on.

Reply:
sure. Filed an issue #481